src/room_data.c

```diff
@@ -4082,7 +4081,27 @@ void kill_room_slab_and_contents(PlayerNumber plyr_idx, MapSlabCoord slb_x, MapS
 
 void free_room_structure(struct Room *room)
 {
-  _DK_free_room_structure(room);
+    PlayerNumber owner = room->owner;
+    if ( game.neutral_player_num != owner )
+    {
+        struct Dungeon *dungeon = get_dungeon(owner);
+
+        if ( room->index == dungeon->room_kind[room->kind] )
+        {
+            dungeon->room_kind[room->kind] = room->next_of_owner;
+            struct Room *next_room = room_get(room->next_of_owner);
+            next_room->prev_of_owner = 0;
+        }
+        else
+        {
+            struct Room *next_room = room_get(room->next_of_owner);
+            next_room->prev_of_owner = room->prev_of_owner;
+            struct Room *prev_room = room_get(room->next_of_owner);
```

Comment:
This looks to me like it should be 
`struct Room *prev_room = room_get(room->prev_of_owner);`

Is it?

Reply:
right yes, don't know how I looked over that, fixed now